plsc/entry.py

```diff
@@ -383,6 +386,8 @@ def build_program(self,
         num_trainers = self.num_trainers
 
         image_shape = [int(m) for m in self.image_shape]
+        if self.data_format == "NHWC":
+            image_shape=[image_shape[1], image_shape[2], image_shape[0]]
         # model definition
```

Comment:
As users can set image shape by set_image_shape, I think it is more reasonable to force users to set the image shape correctly when using mixed precision training. But we can also determine whether users set this parameter correctly, because the channel dimension is usually 3, much smaller than the width and height dimensions.